### PR TITLE
Added the Python Example for the Delete Results API Only

### PR DESCRIPTION
**Pull request by Madhan-Reddy-ni**
**Justification**
In order to represent how to use the APIs created in the TestMonitor service in the SystemLink Enterprise version, there is a need to create examples that will show how to call the APIs, and what details are sent in the API request body.

**Implementation**
In this example, I have demonstrated how to use the Delete Result API and Delete Results API in Python

**Testing**
Manual testing completed

### CODE REVIEW

File: Python Examples/TestMonitor/DeleteResults/delete_results.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    print("Press enter to delete the result")
+    input()
+    delete_single_result(test_result["id"])
+
+
+def create_multiple_results() -> List:
+    test_result = get_test_result()
+    result_ids = []
+    for i in range(0,5):
+        test_result['programName'] = test_result['programName'] + str(i)
```
Comment (mohan-ni) [inline Python Examples/TestMonitor/DeleteResults/delete_results.py]: Is there any reason for creating results under unique program name?

And this results in appending the iteration number like this
image

File: Python Examples/TestMonitor/test_data_manager_client.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        raise ValueError("Missing required parameter 'result_id' when calling ResultsApi->DeleteResultV2")
+    request_uri = f"{base_uri}{delete_result_route}/{result_id}?deleteSteps{delete_steps}"
+    raise_delete_request(request_uri)
+
+def delete_results(result_ids: List, delete_steps: bool = True) -> Dict:
+    """
+    Deletes the existing test results.
+    :param result_ids: result ids which needs to be deleted
+    """
+    if len(result_ids) == 0 or result_ids == None:
```
Comment (mohan-ni) [inline Python Examples/TestMonitor/test_data_manager_client.py]: If `result_ids` is `None`, then checking its length will throw an exception, so we have to check for nullability first

**Review by mohan-ni — APPROVED** [review]

File: Python Examples/TestMonitor/test_data_manager_client.py (hunk 2 above)
Comment (Madhan-Reddy-ni) [reply Python Examples/TestMonitor/test_data_manager_client.py]: updated to check for the nullability first and then for the length.

File: Python Examples/TestMonitor/DeleteResults/delete_results.py (hunk 1 above)
Comment (Madhan-Reddy-ni) [reply Python Examples/TestMonitor/DeleteResults/delete_results.py]: The reason for giving the unique program name is to show unique results in the application. I have updated the logic to append the iteration number to the default program name but not to the previous test result program name.

File: Python Examples/TestMonitor/DeleteResults/ReadMe.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+For example: `python delete_results.py --server https://my_server apiKey`.
+
+How to generate API key
+-----------------------
+Please refer to this [link](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/creating-an-api-key.html) for generating the API key
+
+About the Example
+-----------------
+
+This example creates a single test result and deletes this created result by using Delete result Api and creates multiple(five) test results and deletes all these multiple results at a time by using delete-results POST Api.
```
Comment (ramganesh-ni) [inline Python Examples/TestMonitor/DeleteResults/ReadMe.md]:
```suggestion
This example creates a single test result and deletes this created result by using Delete result Api. Further, it creates multiple(five) test results and deletes all the created results at a time by using delete-results POST Api.
```

File: Python Examples/TestMonitor/DeleteResults/delete_results.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    return "error" in response.keys()
+
+
+def create_single_result() -> Dict:
+    test_result = get_test_result()
+
+    # create test result
+    response = test_data_manager_client.create_results(results=[test_result])
+    if is_partial_success_response(response):
+        raise Exception("Error occurred while creating new test result, please check for correct test result details and have correct access for creating the new test results")
```
Comment (ramganesh-ni) [inline Python Examples/TestMonitor/DeleteResults/delete_results.py]:
```suggestion
        raise Exception("Error occurred while creating new test result, please ensure test result is in a valid format and you have permissions to create test results")
```

Comment (ramganesh-ni) [reply Python Examples/TestMonitor/DeleteResults/delete_results.py]: Please update other messages also accordingly

File: Python Examples/TestMonitor/DeleteResults/delete_results.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    except:
+        print("Error occurred while deleting the test result, please check for the correct result Id and you have correct access for deleting the results")
+
+
+def create_and_delete_single_result() -> None:
+    
+    # create test result
+    print("Creating New test result")
+    test_result = create_single_result()    
+    print(f"Test result has been created under part number={test_result['partNumber']} with Id = {test_result['id']}")
```
Comment (ramganesh-ni) [inline Python Examples/TestMonitor/DeleteResults/delete_results.py]:
```suggestion
    print(f"Test result created under part number={test_result['partNumber']} with Id = {test_result['id']}")
```

File: Python Examples/TestMonitor/DeleteResults/delete_results.py [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+def create_multiple_results() -> List:
+    test_result = get_test_result()
+    program_name = test_result["programName"]
+    result_ids = []
+    for i in range(0,5):
+        test_result["programName"] = program_name + str(i)
+        response = test_data_manager_client.create_results(results=[test_result])
+        if is_partial_success_response(response):
+            print("Error occurred while creating new test result, please check for correct test result details and have correct access for creating the new test results")
```
Comment (ramganesh-ni) [inline Python Examples/TestMonitor/DeleteResults/delete_results.py]:
```suggestion
            print("An error occurred while creating test results. Please ensure the test results are in a valid format or you have permission to create test results")
```

File: Python Examples/TestMonitor/DeleteResults/delete_results.py [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+    if len(result_ids) > 0 :
+        return result_ids
+    else:
+        raise Exception("Error occurred while creating multiple new test results, please check whether you have correct access for creating the new test results")
+
+
+def delete_multiple_results(result_ids: List) -> None:
+    response = test_data_manager_client.delete_results(result_ids, True)
+    if is_partial_success_response(response) :
+        print("Error occurred while deleting the multiple test results, please check for the correct result Ids and you have correct access for deleting the results")
```
Comment (ramganesh-ni) [inline Python Examples/TestMonitor/DeleteResults/delete_results.py]:
```suggestion
        print("Error occurred while deleting test results, please check if the result Ids are valid and you have permissions to delete the results")
```

File: Python Examples/TestMonitor/DeleteResults/delete_results.py [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+    else:
+        print("\nMultiple results has been deleted successfully")
```
Comment (ramganesh-ni) [inline Python Examples/TestMonitor/DeleteResults/delete_results.py]:
```suggestion
        print("\Results have been deleted successfully")
```

File: Python Examples/TestMonitor/DeleteResults/delete_results.py [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+
+
+def create_and_delete_multiple_results() -> None:
+
+    # create multiple test results
+    print("\nCreating multiple test results.\nResult Ids are listed down below")
```
Comment (ramganesh-ni) [inline Python Examples/TestMonitor/DeleteResults/delete_results.py]:
```suggestion
    print("\nCreating multiple test results.\nResult Ids are listed below")    
```

File: Python Examples/TestMonitor/DeleteResults/delete_results.py [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+    else:
+        print("\nMultiple results has been deleted successfully")
+
+
+def create_and_delete_multiple_results() -> None:
+
+    # create multiple test results
+    print("\nCreating multiple test results.\nResult Ids are listed down below")    
+    result_ids = create_multiple_results()    
+    print("\nMultiple test results has been created successfully")
```
Comment (ramganesh-ni) [inline Python Examples/TestMonitor/DeleteResults/delete_results.py]:
```suggestion
    print("\Test results created successfully")
```

File: Python Examples/TestMonitor/DeleteResults/delete_results.py (hunk 6 above)
Comment (mohan-ni) [inline Python Examples/TestMonitor/DeleteResults/delete_results.py]:
```suggestion
            print("Error occurred while creating a new test result, please check the result details and retry.")
```

File: Python Examples/TestMonitor/DeleteResults/delete_results.py [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+    input()
+    delete_multiple_results(result_ids)
+
+@click.command()
+@click.option("--server", help = "Enter server url")
+@click.argument("api_key")
+def main(server, api_key):
+    """
+    To run the example against a SystemLink Enterprise, the URL should include
+    the scheme, host, and port if not default.\n
```
Comment (ramganesh-ni) [inline Python Examples/TestMonitor/DeleteResults/delete_results.py:119]: Why do add these `\n` line breaks here?

File: Python Examples/TestMonitor/DeleteResults/delete_results.py [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+        if is_partial_success_response(response):
+            print("Error occurred while creating new test result, please check for correct test result details and have correct access for creating the new test results")
+        else:
+            test_result = response["results"][0]
+            result_ids.append(test_result["id"])
+            print(f"{test_result['id']}")
+    if len(result_ids) > 0 :
+        return result_ids
+    else:
+        raise Exception("Error occurred while creating multiple new test results, please check whether you have correct access for creating the new test results")
```
Comment (mohan-ni) [inline Python Examples/TestMonitor/DeleteResults/delete_results.py]:
```suggestion
        raise Exception("Error occurred while creating the test results, please check whether you have access for creating new test results")
```

File: Python Examples/TestMonitor/DeleteResults/delete_results.py (hunk 4 above)
Comment (mohan-ni) [inline Python Examples/TestMonitor/DeleteResults/delete_results.py]:
```suggestion
        raise Exception("Error occurred while creating the test result, please check the result details.")
```

File: Python Examples/TestMonitor/DeleteResults/delete_results.py [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+
+    return test_result
+
+
+def delete_single_result(result_id: str) -> None:
+    try:
+        test_data_manager_client.delete_result(result_id, True)
+        print(f"\nThe test result with Id = {result_id} has been deleted")
+    except:
+        print("Error occurred while deleting the test result, please check for the correct result Id and you have correct access for deleting the results")
```
Comment (mohan-ni) [inline Python Examples/TestMonitor/DeleteResults/delete_results.py]:
```suggestion
        print("Error occurred while deleting the test result, please check the result Id and access for deleting the results")
```

File: Python Examples/TestMonitor/DeleteResults/delete_results.py (hunk 7 above)
Comment (mohan-ni) [inline Python Examples/TestMonitor/DeleteResults/delete_results.py]:
```suggestion
        print("Error occurred while deleting the test results, please check the result Ids and access for deleting the results")
```

File: Python Examples/TestMonitor/DeleteResults/delete_results.py [hunk 14]
```diff
@@ -0,0 +1,148 @@
...
+    print("Please enter to delete these results")
+    input()
+    delete_multiple_results(result_ids)
+
+@click.command()
+@click.option("--server", help = "Enter server url")
+@click.argument("api_key")
+def main(server, api_key):
+    """
+    To run the example against a SystemLink Enterprise, the URL should include
```
Comment (mohan-ni) [inline Python Examples/TestMonitor/DeleteResults/delete_results.py]:
```suggestion
    To run the example against SystemLink Enterprise, the URL should include
```

**Review by mohan-ni — APPROVED** [review]

File: Python Examples/TestMonitor/test_data_manager_client.py [hunk 15]
```diff
@@ -0,0 +1,183 @@
...
+    Delete multiple test results.
+    :param result_ids: List of result IDs to be deleted
+    :return: json response after deleting the results
+    """
+    if result_ids == None or len(result_ids) == 0:
+        raise ValueError("result_ids is a required property for ResultsApi->DeleteResultsV2 and cannot be null or empty.")
+    request_url = f"{base_url}{delete_results_route}"
+    body = delete_results_request(result_ids, delete_steps)   
+    request_response = raise_post_request(request_url, body)
+    if request_response.status_code != 204 :
```
Comment (ramganesh-ni) [inline Python Examples/TestMonitor/test_data_manager_client.py:157]: What about a partial success response for Delete, can we handle it as we did in Create Examples?

**Review by ramganesh-ni — APPROVED** [review]

File: Python Examples/TestMonitor/DeleteResults/delete_results.py (hunk 11 above)
Comment (Madhan-Reddy-ni) [reply Python Examples/TestMonitor/DeleteResults/delete_results.py:119]: we used the `\n` to add the line breaks. The click command is not printing the document as written. Instead, it is adding all the lines by adding them to a single line and giving it as output which we don't want. So we are using `\n` to add the line breaks.

**Pull request merged**